src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parser {
+    private final Map<String, ThrowingBiFunction<TaskList, String, String>> functions = new HashMap<>() {
+        {
```

Comment:
I'm not too sure if this is violating Egyptian style! Feel free to let me know if it's not!